check_dependencies.sh

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env bash
+
+# This tool updates dependencies at third_party/maven with dependencies.yaml.
+# If run from CircleCI, it prints out an error if dependencies are not synced.
+
+RED=$(tput setaf 1)
+GREEN=$(tput setaf 2)
+RESET=$(tput sgr0)
+
+BAZEL_DEPS_CACHE="$HOME/bazel_deps_cache"
+DEP_CHECKSUM=($(sha1sum $(pwd)/dependencies.yaml))
+CACHED_RESULT="$BAZEL_DEPS_CACHE/$DEP_CHECKSUM"
+
+if [[ ! -z "$CIRCLECI" ]]; then
+  mkdir -p $BAZEL_DEPS_CACHE
+  isok=$(<$CACHED_RESULT)
+
+  if [ "$isok" == "OK" ]; then
```

Comment:
This cache mechanism allows to "come back" to previously cached `dependencies.yaml`, even if they don't match the current `third_party/maven`.
I think what we need is to somehow figure out if `dependencies.yaml` was changed in the current PR and if so, rerun this. Not sure how to find that out though..

Alternatively, we could either:
* Run it always
* Check the whitelist for dependencies in https://github.com/google/startup-os/blob/master/third_party/maven/workspace.bzl. This doesn't check the correspondence, but at least checks the whitelist.

Reply:
@oferb 
>figure out if `dependencies.yaml` was changed in the current PR and if so, rerun this

maybe `git diff` to `master` may help

Reply:
Actually, looking at `workspace.bzl` I think it'll be pretty easy to do a whitelist check on that.
If `git diff` on `master` works then great, but looking at `workspace.bzl` I think we should do both. They do different things.